packages/react/src/components/Button/ButtonGroup.tsx

```diff
@@ -76,14 +77,16 @@ class ButtonGroup extends UIComponent<ReactProps<ButtonGroupProps>, any> {
 
     return (
       <ElementType {...unhandledProps} className={classes.root}>
-        {_.map(buttons, (button, idx) =>
-          Button.create(button, {
-            defaultProps: {
-              circular,
-              styles: this.getStyleForButtonIndex(styles, idx === 0, idx === buttons.length - 1),
-            },
-          }),
-        )}
+        <Flex gap="gap.smaller">
```

Comment:
@kuzhelov this means introducing an extra DOM node, any suggestions on how to avoid this?
I can think about:
- adding a `marginRight` variable and override it for `Button` when needed
- or specifying
```
defaultProps: {
  as: Flex,
}

// ...

renderComponent() {
  // ...
  return <ElementType gap="gap.smaller" {...unhandledProps} className={classes.root} ... />
}
```

Reply:
yes, I would expect the second approach being used here as, essentially, this is what we would like to achieve - i.e. to use `Flex` as a container.

However, this will introduce one subtle problem (that, depending on the use case, might become not that subtle to spot:). Suppose that client wants to achieve the following
- use <span> as wrapper (this is the place where example seems to be contrived, but I am pretty sure that there might be more appealing scenario)
- preserve layout aspects

Note that this use case won't be covered even with the second approach suggested - however, we might use its core idea being a bit extended. To cover mentioned use case I would suggest to use the following implementation strategy:

```jsx
renderComponent({ ElementType, classes, unhandledProps, ... }) {
  ...
  return <Flex as={ElementType} gap="gap.smaller" {...unhandledProps} className={classes.root} ... />
}
```

Reply:
sounds good, applied `Flex as={ElementType} gap="gap.smaller"`, thanks